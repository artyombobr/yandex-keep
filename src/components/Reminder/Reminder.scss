.reminder {
  display: flex;
  align-items: center;
  margin: 12px 26px;

  &__text {
    font-size: 18px;
    line-height: 24px;
    color: #fff;
    margin-left: 14px;
  }
}
