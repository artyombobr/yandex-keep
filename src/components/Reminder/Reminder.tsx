import * as React from 'react';
import moment from 'moment';
import 'moment/locale/ru';
import './Reminder.scss';

import { ReactComponent as Warning } from '../Note/svg/warning.svg';

function Reminder(reminder: any) {
  return (
    <div className="reminder">
      <Warning />
      <div className="reminder__text">{moment(reminder).fromNow()}</div>
    </div>
  );
}

export default Reminder;
