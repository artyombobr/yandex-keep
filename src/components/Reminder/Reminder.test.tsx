import React from 'react';
import { shallow } from 'enzyme';
import Reminder from './Reminder';

describe('Reminder', () => {
  it('correct render', () => {
    const component = shallow(<Reminder />);
    expect(component).toMatchSnapshot();
  });
});
