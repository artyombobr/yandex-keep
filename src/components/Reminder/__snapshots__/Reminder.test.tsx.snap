// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reminder correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reminder />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef />,
        <div
          className="reminder__text"
        >
          несколько секунд назад
        </div>,
      ],
      "className": "reminder",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "несколько секунд назад",
          "className": "reminder__text",
        },
        "ref": null,
        "rendered": "несколько секунд назад",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef />,
          <div
            className="reminder__text"
          >
            несколько секунд назад
          </div>,
        ],
        "className": "reminder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "несколько секунд назад",
            "className": "reminder__text",
          },
          "ref": null,
          "rendered": "несколько секунд назад",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
